components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java

```diff
@@ -1648,7 +1652,24 @@ public static String resolveUserIdFromUsername(int tenantId, String userStoreDom
                         if (log.isDebugEnabled()) {
                             log.debug("User id could not be resolved for username: " + username + " in user store " +
                                     "domain: " + userStoreDomain + " and tenant with id: " + tenantId + ". Probably " +
-                                    "user does not exist in the user store.");
+                                    "user does not exist in the user store. Hence trying to resolve user from the " +
+                                    "resident organization.");
+                        }
+                        Tenant tenant = IdentityCoreServiceDataHolder.getInstance().getRealmService()
+                                .getTenantManager().getTenant(tenantId);
+                        if(tenant == null) {
+                            return userId;
+                        }
+                        String associatedOrganizationUUID = tenant.getAssociatedOrganizationUUID();
+                        if (associatedOrganizationUUID != null) {
+                            // Trying to resolve the user from the Resident Organization if the user is not found in
+                            // the user store.
```

Comment:
if the user is not found in the user store of the current organization.

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5184/commits/01717332471b2c20636862ea3ff943be0905c224